### ISSUE TITLE
Add option to delete the default network when creating project

### ISSUE THREAD

**Issue by rochdev** [terraform-providers/terraform-provider-google#580]
When using `google_project`, a default network is created by GCP. Since most production workloads probably don't need this network, it would be nice to have an option on `google_project` to disable the default network.

According to the [documentation](https://cloud.google.com/compute/docs/vpc/#vpc_networks_and_projects), it is safe to delete it and even recommended to release one slot from the quota.

**Comment by rochdev** [comment]
This can be done with a provisioner

**Comment by rochdev** [comment]
Reopening since using `null_resource` is generally discouraged.

**Comment by ndmckinley** [comment]
I think the correct approach from our perspective is to use a provisioner for this.  This is what I did, and it seemed to work (although the mixing of gcloud and terraform makes me feel pretty :upside_down_face:, and it requires some environment configuration to make sure they're using the same credentials, which could be improved)

```
resource "random_id" "project" {
  byte_length = 8
}

resource "google_project" "project" {
    project_id = "tf-test-${random_id.project.hex}"
    name = "Terraform source repo"
    org_id = "${var.org}"
    billing_account = "${var.billing_account}"
}

resource "google_project_service" "service" {
  project = "${google_project.project.project_id}"
  service = "compute.googleapis.com"

  provisioner "local-exec" {
    command = "gcloud -q compute networks delete default --project=${self.project}"
  }
}
```

I don't think we're likely to attach this feature to the provider in the near future - we're trying to hew closely to the API where possible - but does the provisioner approach help?

**Comment by sryabkov** [comment]
I wanted to post this in case somebody else runs into this thread first. 

This issue was addressed in #1312, you can now use `auto_create_network` to disable the creation of default network when creating the project. The following works for me with TF `0.11.7` and google cloud provider `1.16`

```
resource "google_project" "project" {
  name                = "${var.project_name}"
  project_id          = "${random_id.random.hex}"
  folder_id           = "${var.folder_id}"
  billing_account     = "${var.billing_account}"
  auto_create_network = false
}
```

See [TF Google Cloud Provider documentation](https://www.terraform.io/docs/providers/google/r/google_project.html) for more info.
